static/js/services/enketo.js

```diff
@@ -315,8 +315,13 @@ angular.module('inboxServices').service('Enketo',
         })
         .then(function(doc) {
           replaceJavarosaMediaWithLoaders(id, doc.html);
-          var form = renderFromXmls(doc, selector, instanceData);
+          return renderFromXmls(doc, selector, instanceData);
+        })
+        .then(function(form) {
```

Comment:
Again, this is just introducing another promise step for no specific reason (except for readability?) as far as I can tell.  Is that correct?

Reply:
Actually no. I realised while looking at this code that `renderFromXmls` is actually async so not having this change could actually cause an issue.

Reply:
So `form` would have previously been a `Promise`, then?  Looks like the old naming was a bit weird, but now makes sense :slightly_smiling_face:

Reply:
Right - I think it used to be a sync function and obviously wasn't updated/renamed when it was changed to async...